### PR TITLE
Add update_device_attrib api

### PR DESCRIPTION
**Pull request by TridTech**
Add an endpoint to the API for modifying attributes such as override_icmp_disable, and override_sysContact_bool.  Supports changing multiple attributes at a time.

Adds the route to the new endpoint and the documentation to support it.

DO NOT DELETE THE UNDERLYING TEXT

#### Please note

> Please read this information carefully. You can run `./lnms dev:check` to check your code before submitting.

- [x] Have you followed our [code guidelines?](https://docs.librenms.org/Developing/Code-Guidelines/)
- [x] If my Pull Request does some changes/fixes/enhancements in the WebUI, I have inserted a screenshot of it.

#### Testers

If you would like to test this pull request then please run: `./scripts/github-apply <pr_id>`, i.e `./scripts/github-apply 5926`
After you are done testing, you can remove the changes with `./scripts/github-remove`.  If there are schema changes, you can ask on discord how to revert.

### CODE REVIEW

File: includes/html/api_functions.inc.php [hunk 1]
```diff
@@ -1844,6 +1844,34 @@ function rename_device(Illuminate\Http\Request $request)
...
+    // use hostname as device_id if it's all digits
+    $device_id = ctype_digit($hostname) ? $hostname : getidbyname($hostname);
+    $data = json_decode($request->getContent(), true);
+    if (empty($data['attrib_type'])) {
+        return api_error(400, 'Device Attrib field to patch has not been supplied');
+    }
+    if (is_array($data['attrib_type']) && is_array($data['attrib_value'])) {
+        if (count($data['attrib_type']) == count($data['attrib_value'])) {
+            for ($x = 0; $x < count($data['attrib_type']); $x++) {
+                if (dbUpdate(['attrib_value' => $data['attrib_value'][$x]], 'devices_attribs', '`device_id`=? and `attrib_type`=?', [$device_id, $data['attrib_type'][$x]]) < 0) {
```
Comment (murrant) [inline includes/html/api_functions.inc.php:1859]: Please don't use the legacy db commands.  https://laravel.com/docs/8.x/eloquent
The Device model also has attrib helpers on it.

File: doc/API/Devices.md [hunk 2]
```diff
@@ -1264,6 +1264,51 @@ Output:
...
+Update devices attributes in the database.
+
+Route: `/api/v0/attribs/:hostname`
+
+- hostname can be either the device hostname or id
+
+Input (JSON):
+
+- attrib_type: The attribute type within the database (can be an array of fields)
+- attrib_value: The attribute value to update the column with (can be an array of data))
```
Comment (murrant) [inline doc/API/Devices.md:1278]: Why not use key value pairs instead of two separate arrays?

File: includes/html/api_functions.inc.php (hunk 1 above)
Comment (TridTech) [reply includes/html/api_functions.inc.php:1859]: To be honest, I started with a copy of the update_device API, and adapted it from there.  The device helpers for attribs is a better spot to work from.  I'll look at re-writing for that.

File: doc/API/Devices.md (hunk 2 above)
Comment (TridTech) [reply doc/API/Devices.md:1278]: The device api uses a field: name, data: value format.  I figured it was best to stay in the style of the existing api.

**Comment by TridTech** [comment]
Cancelling/closing for the time being until I have time to refactor to newer methods.

**Pull request closed without merging**